### PR TITLE
chore: add e2e tests for `GET /items/:id`, `GET /account`, `GET /dashboard`, `GET /dashboard/stats` and `GET /dashboard/users`

### PR DESCRIPTION
**Pull request by revgum**
Adding more coverage on behalf of #524

### CODE REVIEW

File: e2e_test.ts [hunk 1]
```diff
@@ -624,3 +454,233 @@ Deno.test("[e2e] GET /notifications/[id]", async (test) => {
...
+    assertFalse((await resp.text()).includes("Sign in to comment"));
+  });
+
+  await test.step("renders the item page directing an anonymous user to sign in to comment", async () => {
+    const resp = await handler(new Request(url));
+    assertStringIncludes(await resp.text(), "Sign in to comment");
+  });
+});
+
+Deno.test("[e2e] GET /account", async (test) => {
```
Comment (revgum) [inline e2e_test.ts:491]: Added missing tests for `/account` but then moved all of the other `/account/*` tests down to be alongside this one.

**Comment by revgum** [comment]
I expect codecov to capture the diff when this is merged.. Locally, using `genhtml` from Lcov shows the updated coverage.
Screenshot 2023-09-10 at 10 51 10 AM

**Comment by revgum** [comment]
Ah I see, codecov ignores all `*.tsx`, so even the root routes won't be captured. 

@iuioiua what do you think about updating `codecov.yml` to just ignore `components` and `islands` instead of all `.tsx` files?

File: e2e_test.ts [hunk 2]
```diff
@@ -298,37 +128,78 @@ Deno.test("[e2e] GET /signout", async () => {
...
+    assertEquals(resp.status, Status.SeeOther);
+  });
+
+  await test.step("renders dashboard stats chart for a user who is signed in", async () => {
+    const resp = await handler(
+      new Request(url, {
+        headers: { cookie: "site-session=" + user.sessionId },
+      }),
+    );
+    assertStringIncludes(await resp.text(), "<!--frsh-chart_default");
```
Comment (revgum) [inline e2e_test.ts:178]: The page itself renders an outlet for the island, and this assert is looking for the first piece of that outlet.

File: e2e_test.ts [hunk 3]
```diff
@@ -298,37 +128,78 @@ Deno.test("[e2e] GET /signout", async () => {
...
+    assertEquals(resp.status, Status.SeeOther);
+  });
+
+  await test.step("renders dashboard stats table for a user who is signed in", async () => {
+    const resp = await handler(
+      new Request(url, {
+        headers: { cookie: "site-session=" + user.sessionId },
+      }),
+    );
+    assertStringIncludes(await resp.text(), "<!--frsh-userstable_default");
```
Comment (revgum) [inline e2e_test.ts:201]: The page itself renders an outlet for the island, and this assert is looking for the first piece of that outlet.

**Comment by revgum** [comment]
Alrighty.. definitely feel free to take this PR and run with it in terms of syntax adjustments or changes here and there. I've got to get back into my regular obligations. It's been a bunch of fun! I'll try to circle back around to this repo and add contributions as I find more time.

**Review by iuioiua — APPROVED** [review]
I'll add these in, but these will be analysed in Codecov after v1. I'd like to focus on REST API endpoints as they involve more logic. Either way, this is great! Thank you, Josh.

**Pull request merged**